main page 개발

haneeeul opened the pull request:
## PR Checklist ☑️

<!-- Please check the one that applies to this PR using "x". -->

- [x] The commit message follows [guidelines](https://github.com/ondal-house/zu-story/wiki/commit-convention)
- [x] ~~Tests for the changes have been added~~
- [ ] Approve by 3 or more people
- [x] ~~Pass Pipeline~~

## What is this PR 🔍
<!-- Please explain the PR.-->
1. 패키지 추가
2. main page 개발

## PR Describe(Changes) 📝
<!-- Please explain what was added or changed in the PR.-->
### 스타일 적용을 위한 패키지 추가
- mui icon 사용을 위한 패키지 추가
- sass 적용을 위한 패키지 추가
- style class 를 편리하게 적용하기 위해 classNames 패키지 추가
  
### main page 개발
- 페이지 전체 레이아웃을 body 기준으로 header, main, footer 로 나누어 정의
  - header : 상단에 고정
  - main : 컨텐츠 높이에 맞춰 스크롤이 생기도록 설정
  - footer : main 아래에 붙도록 설정
- main 영역을 공통적으로 잡기 위해서 Layout component 정의
- main page 내부 컨텐츠 개발
  - BE 연결 없이 레이아웃 확인을 위해 테스트 데이터 정의
  - 테스트 데이터 기반으로 화면 작성

## How To Test ☑️
<!-- Please explain how to test the PR.-->
```
git remote update && git checkout feature/main-page
npm install
npm start
```

http://localhost:5173/ 접속하여 화면 확인

haneeeul committed:
chore : add necessary packages

스타일 적용을 위해 필요한 패키지를 추가합니다.
mui icon 사용을 위한 패키지
style class 적용을 위한 classNames 패키지
sass 패키지

haneeeul committed:
feat : modify test and main page

test 페이지를 healthcheck 페이지로 수정하고 url 을 /test 로 수정

main 페이지를 수정하고 url 을 / 로 수정
Layout 컴포넌트 사용하여 기본 레이아웃 정의
  - header, main, footer 영역 정의
    - header: 상단 고정
    - footer: main 길이에 따라 하단에 붙도록 개발
  - desktop 디자인이 완성되지 않아 tablet view width 를 최대 너비로
    작업

css width 작업을 편하게 하기 위해 box-sizing 기본값을 border-box 로 수정

haneeeul committed:
feat : create main content

main page 에서 사용하는 category, beverage, board 목록 임시 데이터 및
컴포넌트 생성

ypd01018 commented:
드디어! 마참내!

ypd01018 commented on src/pages/HomePage/BeverageContentContainer.tsx:
```diff
@@ -0,0 +1,33 @@
+  const [errorCode, setErrorCode] = useState<number>();
+  const [errorMessage, setErrorMessage] = useState<string>();
+
+  const { data: beverageList } = useQuery('beverages', API.getBeverages, {
```
useQuery에서 key를 api url과 일치시키면 좋을 것 같아요.
추후 아마` /beaverages` 이런식으로 key를 사용하고 상세 조회 시 `/beverages/${beverageId}`로 사용함으로서 서로 key는 다르지만 data를 일치시키는 방식을 사용할 것 같다고 설계하고 있었어요!

haneeeul replied:
좋은 생각!  
그럼 post /beverages/${beverageId} <-> get /beverages/${beverageId} 구분을 위해 key 값은  
http method + url 조합으로 하겠습니다.

haneeeul replied:
`get/beverages`  
이렇게 key 설정했는데 더 좋은 의견이 있을까요?  
@Simeuilyang @gychoi0916 @ResinCRT

ypd01018 replied:
get 안해도 돼요 ㅋㅋ
조회성인 HTTP GET Method를 사용하는 경우 useQuery를 통해 Cache에 적재, 
삽입/수정/삭제의 경우 useMutation을 통해 Cache에 key:data쌍으로 저장된 데이터를 조작합니다.

ypd01018 replied:
그리고 저도 지금 생각난건데, React Query는 Key를 배열로 관리하더라구요? 이유는 잘 몰라요 찾아봐야 할듯 ㅋㅋ
그래서 ['beverages']로 하면 될 것 같아요.
https://www.zigae.com/react-query-key/

ypd01018 commented on src/tempData/data.ts:
```diff
@@ -0,0 +1,200 @@
+export interface board {
```
요 부분 중 계속 유지 될 친구들 ex(api response, beverage , category  ... 등등)은 model 디렉터리를 따로 만들어서 분류해도 좋을 것 같아요. tempData의 디렉터리가 옳은가에서 약간 애메한 감이 보여요!

haneeeul replied:
tempData 에 인터페이스랑 임시 데이터를 같이 밀어넣은건 BE DTO 가 확정되면 옮기려고 한 것이었는데.. 일단 선 분리를 하겠습니다.

haneeeul committed:
feat : modify useQuery key to 'http method + url'

useQuery에서 key를 api method + url 로 변경하여
추후에 data format 은 일치되더라도 key 는 독립적으로 사용 가능하도록 함.

haneeeul committed:
refactor : add model directory

모델 디렉토리를 만들어 interface 를 임시 데이터와 분리함.
